### PR TITLE
Inline Help: Remove Checklist prompt

### LINKED ISSUE #40384: Remove Checklist Notification ViewMode from Inline Help
Let's remove this site setup checklist notice from inline help, when we create a new site. Please also delete unused code from the checklist component.

Screen Shot 2020-03-23 at 4 18 34 PM

### PR DESCRIPTION
**Pull request by sixhours**
#### Changes proposed in this Pull Request

* I giveth, and I taketh away. 😂 Removes the checklist "prompt" feature from inline help. Now that My Home is the default landing page, it's redundant.
* Removes checklist markup from the Checklist component itself.
* Removes related redux values for showing the checklist notification and tasks.

**Before**

Screen Shot 2020-03-24 at 3 50 27 PM

**After**

Screen Shot 2020-03-24 at 3 50 11 PM

#### Testing instructions

* Switch to this PR
* Navigate around Calypso and confirm the checklist prompt does not appear in Inline Help
* Make sure there are no errors in the console as a result of the removed code

Fixes #40384

### CODE REVIEW

**Comment by gwwar** [comment]
:D sorry needs a rebase after my PR landed. The red line count here is excellent though @sixhours

File: client/state/inline-help/actions.js [hunk 1]
```diff
@@ -146,22 +142,6 @@ export function hideInlineHelpPopover() {
 	};
 }
 
-export function showChecklistPrompt() {
```
Comment (mmtr) [inline client/state/inline-help/actions.js]: I found these actions are still used in other parts of the code:

https://github.com/Automattic/wp-calypso/blob/dc580cf2b9ae3195ece6a7381aba03165ef11baf/client/my-sites/preview/main.jsx#L25-L29

Not sure if we need to maintain the linked code or if we can remove it too as part of this PR.

File: client/state/inline-help/actions.js [hunk 2]
```diff
@@ -146,22 +142,6 @@ export function hideInlineHelpPopover() {
...
-	return dispatch => {
-		dispatch( {
-			type: INLINE_HELP_CHECKLIST_PROMPT_SHOW,
-		} );
-	};
-}
-
-export function hideChecklistPrompt() {
```
Comment (mmtr) [inline client/state/inline-help/actions.js]: https://github.com/Automattic/wp-calypso/blob/dc580cf2b9ae3195ece6a7381aba03165ef11baf/client/my-sites/checklist/wpcom-checklist/checklist-prompt/task.jsx#L16-L20

File: client/state/inline-help/actions.js [hunk 3]
```diff
@@ -177,21 +157,3 @@ export function hideOnboardingWelcomePrompt() {
 		} );
 	};
 }
-
-export function setChecklistPromptTaskId( taskId ) {
```
Comment (mmtr) [inline client/state/inline-help/actions.js]: https://github.com/Automattic/wp-calypso/blob/dc580cf2b9ae3195ece6a7381aba03165ef11baf/client/my-sites/checklist/wpcom-checklist/checklist-prompt/task.jsx#L16-L20

File: client/state/inline-help/actions.js [hunk 4]
```diff
@@ -177,21 +157,3 @@ export function hideOnboardingWelcomePrompt() {
...
-	return dispatch => {
-		dispatch( {
-			type: INLINE_HELP_CHECKLIST_PROMPT_SET_TASK_ID,
-			taskId,
-		} );
-	};
-}
-
-export function setChecklistPromptStep( step ) {
```
Comment (mmtr) [inline client/state/inline-help/actions.js]: https://github.com/Automattic/wp-calypso/blob/dc580cf2b9ae3195ece6a7381aba03165ef11baf/client/my-sites/checklist/wpcom-checklist/checklist-prompt/task.jsx#L16-L20

File: client/state/inline-help/selectors.js [hunk 5]
```diff
@@ -135,17 +125,3 @@ export function isInlineHelpChecklistPromptVisible( state ) {
...
 	return get( state, 'inlineHelp.onboardingWelcomePrompt.isVisible', false );
 }
-
-/**
- * Returns the task ID that should show up in the inline help popover.
- *
- * @param  {object}  state  Global state tree
- * @returns {string}         The task ID
- */
-export function getChecklistPromptTaskId( state ) {
```
Comment (mmtr) [inline client/state/inline-help/selectors.js]: These selectors are still used too:

https://github.com/Automattic/wp-calypso/blob/dc580cf2b9ae3195ece6a7381aba03165ef11baf/client/my-sites/checklist/wpcom-checklist/checklist-prompt/index.jsx#L15

File: client/state/inline-help/selectors.js [hunk 6]
```diff
@@ -135,17 +125,3 @@ export function isInlineHelpChecklistPromptVisible( state ) {
...
-	return get( state, 'inlineHelp.checklistPrompt.taskId', null );
-}
-
-export function getChecklistPromptStep( state ) {
```
Comment (mmtr) [inline client/state/inline-help/selectors.js]: https://github.com/Automattic/wp-calypso/blob/dc580cf2b9ae3195ece6a7381aba03165ef11baf/client/my-sites/checklist/wpcom-checklist/checklist-prompt/task.jsx#L21

**Comment by sixhours** [comment]
@mmtr Great catch! I had to do some digging into what `ChecklistPrompt` actually was because I didn't remember, and I didn't comment my code. :P 

At one point after completing signup, we sent folks to a `?checklist` route on the site previewer that popped up inline help with a prompt of their next checklist task. That's no longer a thing, so I've removed it entirely.

It appears there's also an "onboarding welcome prompt" that is similar, but I don't know if that's currently in use so I'm leaving it alone for now.

**Comment by gwwar** [comment]
I think we should be able to get rid of the "notification" view type:

https://github.com/Automattic/wp-calypso/blob/8ccdbe3daabe3c78dddba0d75de38f2fc85c9d61/client/my-sites/checklist/wpcom-checklist/index.jsx#L26

**Pull request merged**